fastapi/app/main.py

```diff
@@ -0,0 +1,231 @@
+"""Module providing SpiceQL endpoints"""
+
+from ast import literal_eval
+from typing import Annotated, Any, Optional
+from fastapi import FastAPI, Query
+from pydantic import BaseModel, Field
+import pyspiceql
+
+
+# Models
+class MessageItem(BaseModel):
+    message: str
+
+class BodyModel(BaseModel):
+    result: Optional[Any] = None
+    error: Optional[str] = None
+
+class ResultModel(BodyModel):
+    result: Any = Field(serialization_alias='return')
+
+class ErrorModel(BodyModel):
+    error: str
+
+class ResponseModel(BaseModel):
+    statusCode: int
+    body: dict | str
+
+# Create FastAPI instance
+app = FastAPI()
+
+# General endpoints
+@app.get("/")
+async def root():
+    return {"message": "Visit the /docs endpoint to see the Swagger UI."}
+
+@app.post("/customMessage")
+async def message(
+    message_item: MessageItem
+    ):
+    return {"message": message_item.message}
+
+
+# SpiceQL endpoints
+@app.get("/getTargetStates")
+async def getTargetStates(
+    ets: Annotated[list[float], Query()] | str,
+    target: str,
+    observer: str,
+    frame: str,
+    abcorr: str,
+    mission: str,
+    ckQuality: str = "",
+    spkQuality: str = "",
+    searchKernels: bool = False):
```

Comment:
Where do these get documented for rendering in the swagger endpoint? For example, is `target` defined somewhere as a string pulled from an Enum?

Reply:
The Swagger UI docs are [automatically generated](https://fastapi.tiangolo.com/features/#automatic-docs), you can see the _openapi.json_ content that is generated at the endpoint `/openapi.json`. The docs generation only seems to extend within the scope of the FastAPI app, so `target` would accept any string. 

If we would like to use enums and have stricter rules on the data model validation, we can definitely do that. One way I can see about going that is, for example, creating a request model, `TargetStatesModel`, that has all those query params as attributes with `target` accepting only `TargetEnum` (custom model). Then, when the docs are generated, `target` would be defined as an enum list instead of any string. Or something like that.

Reply:
Cool - whatever is cool here. I'm just learning the ropes. I've gone from a monkey-patch all the things to typing as much as is reasonable, especially if it helps with documentation perspective as I work on more standards/specifications. 

I think if you all decide that this is going to become a spec that you hope others might adhere to, then more description of the models is a positive. If that isn't the goal, the added description might just be added work that doesn't add a lot.